File: Baekjoon/2309_일곱난쟁이.java
/*
import java.util.Arrays;
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int arr[] = new int[9];
		int ans[] = new int[7];
		int result=0;
		
		for(int i=0; i<9; i++)
			arr[i]= scan.nextInt();
		
		
		Arrays.sort(arr);
		for(int i=0; i<9; i++) {
			for(int j=i; j<9; j++) {
				for(int k=0; k<9; k++) {
					if(k!=i && k!=j)
						result+=arr[k];
				}
				if(result==100) {
					for(int k=0; k<9; k++) {
						if(k!=i && k!=j) {
							System.out.println(arr[k]);
						}
					}
				}
				result=0;
			}
		}
		
	}
}
*/

File: Baekjoon/2309_일곱난쟁이.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int bj_2309() {
	vector<int> arr(9);
	vector<int> select(9, 0);
	for (int i = 0; i < 9; ++i)
		cin >> arr[i];

	sort(arr.begin(), arr.end());

	select[8] = 1;
	select[7] = 1;

	do {
		int result = 0;
		for (int i = 0; i < 9; ++i) {
			if (select[i] == 0)
				result += arr[i];
		}

		if (result == 100)
			break;
	} while (next_permutation(arr.begin(), arr.end()));

	for (int i = 0; i < 9; ++i) {
		if (select[i] == 0)
			cout << arr[i] << '\n';
	}
	return 0;
}
